src/btree/bt_curnext.c

```diff
@@ -595,32 +594,19 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
 	F_CLR(cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET);
 
 	/*
-	 * When retrying an operation due to a prepare conflict, the cursor is
-	 * is at an update list which resulted in conflict. So, when retrying
-	 * we should examine the same update again instead of iterating to the
-	 * next object. We'll eventually find a valid update, return prepare-
-	 * conflict until successful.
+	 * If this cursor has returned prepare conflict earlier, check to see
+	 * whether that prepared update is resolved or not. If not resolved,
+	 * continue returning prepare conflict. If resolved, return the value
+	 * based on the visibility rules.
 	 */
-	F_CLR(cbt, WT_CBT_RETRY_PREV);
-	if (F_ISSET(cbt, WT_CBT_RETRY_NEXT)) {
-		WT_ERR(__wt_cursor_valid(cbt, &upd, &valid));
-		if (!valid)
-			WT_ERR(WT_PREPARE_CONFLICT);
-
-		/* The update that returned prepared conflict is now visible. */
-		F_CLR(cbt, WT_CBT_RETRY_NEXT);
-
-		/*
-		 * The underlying key-return function uses a comparison value
-		 * of 0 to indicate the search function has pre-built the key
-		 * we want to return. That's not the case, don't take that path.
-		 */
-		cbt->compare = 1;
-		WT_ERR(__cursor_kv_return(session, cbt, upd));
+	if (F_ISSET(cbt, WT_CBT_ITERATE_RETRY_NEXT)) {
+		WT_ERR(__cursor_check_prepared_update(cbt, &visible));
+		if (visible) {
 #ifdef HAVE_DIAGNOSTIC
-		WT_ERR(__wt_cursor_key_order_check(session, cbt, true));
+			ret = __wt_cursor_key_order_check(session, cbt, true);
```

Comment:
It's unusual for us to ignore an error return code - even from a diagnostic function. If we are going to ignore it we have a macro: `WT_IGNORE_RET`. Could we handle this somehow?